python/dglke/models/pytorch/ke_tensor.py

```diff
@@ -0,0 +1,308 @@
+# -*- coding: utf-8 -*-
+#
+# tensor_models.py
+#
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+KG Sparse embedding
+"""
+import os
+import numpy as np
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as functional
+import torch.nn.init as INIT
+
+import torch.multiprocessing as mp
+from torch.multiprocessing import Queue
+from _thread import start_new_thread
+import traceback
+from functools import wraps
+
+from .. import *
+
+from .tensor_models import thread_wrapped_func
+
+@thread_wrapped_func
+def async_update(num_thread, emb, queue):
```

Comment:
i don't understand why the code is moved here?

Reply:
I just copy paste to here. Maybe we can move them when doing training code refactor.